pyscriptjs/src/components/pyscript.ts

```diff
@@ -1,117 +1,37 @@
-import {
-    addToScriptsQueue,
-} from '../stores';
-
-import { getAttribute, addClasses, htmlDecode } from '../utils';
-import { BaseEvalElement } from './base';
+import { htmlDecode, ensureUniqueId } from '../utils';
 import type { Runtime } from '../runtime';
 import { getLogger } from '../logger';
+import { pyExec } from '../pyexec';
 
 const logger = getLogger('py-script');
 
-export class PyScript extends BaseEvalElement {
-    constructor() {
-        super();
-
-        // add an extra div where we can attach the codemirror editor
-        this.shadow.appendChild(this.wrapper);
-    }
-
-    connectedCallback() {
-        this.checkId();
-        this.code = htmlDecode(this.innerHTML);
-        this.innerHTML = '';
-
-        const mainDiv = document.createElement('div');
-        addClasses(mainDiv, ['output']);
-        // add Editor to main PyScript div
-
-        const output = getAttribute( this, "output");
-        if (output) {
-            const el = document.getElementById(output);
-            if( el ){
-                this.errorElement = el;
-                this.outputElement = el;
-            }
-
-            // in this case, the default output-mode is append, if hasn't been specified
-            if (!this.hasAttribute('output-mode')) {
-                this.setAttribute('output-mode', 'append');
-            }
-        } else {
-            const stdOut = getAttribute( this, "std-out");
-            if (stdOut) {
-                const el = document.getElementById( stdOut );
-                if( el){
-                    this.outputElement = el
-                }
-            } else {
-                // In this case neither output or std-out have been provided so we need
-                // to create a new output div to output to
-
-                // Let's check if we have an id first and create one if not
-                this.outputElement = document.createElement('div');
-                const exec_id = this.getAttribute('exec-id');
-                this.outputElement.id = this.id + (exec_id ? '-' + exec_id : '');
-
-                // add the output div id if there's not output pre-defined
-                mainDiv.appendChild(this.outputElement);
-            }
-
-            const stdErr = getAttribute( this, "std-err");
-            if ( stdErr ) {
-                const el = document.getElementById( stdErr );
-                if( el ){
-                    this.errorElement = el;
-                }else{
-                    this.errorElement = this.outputElement;
-                }
-            } else {
-                this.errorElement = this.outputElement;
-            }
-        }
+export function make_PyScript(runtime: Runtime) {
```

Comment:
Just curious - what's the advantage of using a function that defines the class rather than just the class definition?

Reply:
the fact that I can pass `runtime` from the outside. Note that `runtime` is used from `connectedCallback`, and since this is a method called by the browser, we don't really have any other way to pass it explicitly.
The alternative was to use a global variable as it was before, but we saw that it caused way too many problems.